src/iter/internal.rs

```diff
@@ -166,6 +172,34 @@ impl Splitter {
     }
 }
 
+#[derive(Clone, Copy)]
+struct LengthSplitter {
+    inner: Splitter,
+    min: usize,
+}
+
+impl LengthSplitter {
+    #[inline]
+    fn new(min: usize, max: usize, len: usize) -> LengthSplitter {
+        let mut splitter = LengthSplitter {
+            inner: Splitter::new(),
+            min: cmp::max(min, 1),
+        };
+
+        let max_splits = len / cmp::max(max, 1);
```

Comment:
Ideally, with a few examples that show how different min/max/len combinations work out